rosbag2_transport/src/rosbag2_transport/qos.cpp

```diff
@@ -19,20 +19,37 @@
 
 #include "qos.hpp"
 
+namespace
+{
+// These values are returned to mean "unspecified" in FastDDS
+// 0 is equivalent and is portable to other implementation
+static constexpr uint64_t FASTDDS_UNSPECIFIED_SEC = 2147483647;
+static constexpr uint64_t FASTDDS_UNSPECIFIED_NSEC = 4294967295;
+}  // namespace
+
 namespace YAML
 {
 Node convert<rmw_time_t>::encode(const rmw_time_t & time)
 {
   Node node;
-  node["sec"] = time.sec;
-  node["nsec"] = time.nsec;
+  if (time.sec == FASTDDS_UNSPECIFIED_SEC && time.nsec == FASTDDS_UNSPECIFIED_NSEC) {
```

Comment:
Is there any possibility that that one of these values is specified and the other is not?

Reply:
It's the specific combination of values that is returned by that implementation to mean "unspecified" - i didn't want to make any assumptions about what happened in other cases.